### ISSUE TITLE
Fine Tuning problem with binary classification project

### ISSUE THREAD

**Issue by KamalOthman** [keras-team/keras#9785]
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- 

- [x] [] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps

- [ ] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

Hi 
I used the below program for fine tuning VGG16 for 5 classes (11600 for each class) following the example in https://keras.io/applications/ . It gives a good validation accuracy, which is around 93%. However, I used the same architecture to solve 2 classes selected from the same 5 classes (binary classifier), the validation accuracy always stays around 42%.
I expected to get better results. 
Is the problem from the model or the data? 
Any advice?

Thanks

Here is the fine-tuning learning code:

```
base_model = VGG16(weights='imagenet', include_top=False)
num_layers = 15
print('NUMBER OF NON-TRAINABLE LAYERS: ',num_layers)

# add a global spatial average pooling layer
x = base_model.output
x = GlobalAveragePooling2D()(x)
# let's add a fully-connected layer
x = Dense(1024, activation='relu')(x)
# and a logistic layer -- let's say we have 200 classes
predictions = Dense(classes_output, activation='softmax')(x)

# this is the model we will train
model = Model(input=base_model.input, output=predictions)

# first: train only the top layers (which were randomly initialized)
# i.e. freeze all convolutional InceptionV3 layers
for layer in base_model.layers:
    layer.trainable = False

# compile the model (should be done *after* setting layers to non-trainable)
#model.compile(optimizer='rmsprop', loss='categorical_crossentropy', ['accuracy'])
from keras.optimizers import Adam
model.compile(optimizer=Adam(lr=0.0001), loss = classification_type, metrics=['accuracy'])

# train the model on the new data for a few epochs
model.fit_generator( gen_train,
                                 steps_per_epoch=len(train_data) // 32,
                                 validation_data = gen_valid,
                                 validation_steps = len(valid_data),
                                 epochs=nb_epochs)

# at this point, the top layers are well trained and we can start fine-tuning
# convolutional layers from inception V3. We will freeze the bottom N layers
# and train the remaining top layers.

# let's visualize layer names and layer indices to see how many layers
# we should freeze:
for i, layer in enumerate(base_model.layers):
   print(i, layer.name)

# we chose to train the top 2 inception blocks, i.e. we will freeze
# the first "num_layers" layers and unfreeze the rest:
for layer in model.layers[:num_layers]:
   layer.trainable = False
for layer in model.layers[num_layers:]:
   layer.trainable = True

# we need to recompile the model for these modifications to take effect
# we use SGD with a low learning rate
from keras.optimizers import SGD
#model.compile(optimizer=SGD(lr=0.0001, momentum=0.9), loss='categorical_crossentropy',['accuracy'])
model.compile(optimizer=SGD(lr=0.0001, momentum=0.9), loss = classification_type, metrics=['accuracy'])
# we train our model again (this time fine-tuning the top 2 inception blocks
# alongside the top Dense layers
history = model.fit_generator( gen_train,
                                 steps_per_epoch=len(train_data) // 32,
                                 validation_data = gen_valid,
                                 validation_steps = len(valid_data),
                                 epochs=nb_epochs)

# serialize model to JSON
model_json = model.to_json()
with open("results_binary/model.json", "w") as json_file:
    json_file.write(model_json)
    
model.save_weights(model_weights_path)
print('******** MODEL & WEIGHTS ARE SAVED *********')
```

**Comment by KamalOthman** [comment]
Dear friends,
I am still waiting for your advices.

Thanks
